### DISCUSSION TITLE
Some help with old program

### DISCUSSION THREAD

**Discussion by BMtech118** [robhagemans/pcbasic#235] [General]
hello, Sorry this isn't normally something id ask but we have a customer with an old program built in GW Basic and having issues with a syntax error on one of the programs, currently they are using Virutal machines to run windows xp and it works perfectly but this one program within pc basic is having issues i was wondering if anyone at all could assist as i just can't get it to work. 

Cheers

**Comment by robhagemans** [comment]
At the risk of getting myself into unpaid consulting work...

What is the syntax error you get? I.e. can you post the code line that causes it?

Do you mean the BASIC program is running fine on GW-BASIC under XP but not on PC-BASIC? Or only tried on PC-BASIC under XP?

Note that PC-BASIC doesn't officially support XP anymore, though it may well work.

**Comment by BMtech118** [comment]
Hey @robhagemans  Thank you so much for your reply and i honeslty have no issue with paid consultancy work for this instance. Essentially running the program via PC BASIC and running this specific program it gives error:

![image](https://github.com/robhagemans/pcbasic/assets/157108548/01ec7f33-ba29-4210-8155-baccb0cb6300)

However when running it via XP it works fine it only actually errors on PC Basic using the same program files. 

Sorry im really probably not explaining this properly but they only use XP as thats what they used to use when this program was first designed for them, and then over the years they have moved to XP mode when on windows 8, then XP again now on windows 10 and 11. 

My idea was to get them to use PC basic to run the programs which works for the other 6 programs they have but when running this one specific program on PC Basic it errors. 

hope this clarifies more?

**Reply by robhagemans** [reply]
Thanks, paid consulting is not an option anyway, don't want to alienate my regular employer.

Not sure how this would work in regular GW-BASIC, as `<> =` is simply not correct syntax. It should be either `<>` (not equals) or `=` (equals) depending on what you're trying to achieve.

If this indeed does run in GW-BASIC, my hypothesis would be that it's stored as a tokenised file and something odd is happening in the binary encoding. You could try to `LIST` this line in GW-BASIC and see if it shows the same way. If it looks correct there, you could try to save in text format `SAVE"name",A` and load the resulting file in PC-BASIC. 

In that case it would be interesting to have the original tokenised file to see what's going on there, as I haven't seen this before.

**Reply by BMtech118** [reply]
That would make sense. 

here is both side by side they are exactly the same. I thought editing the file itself would be fine but it doesnt seem to save the file so im clearly doing it wrong 
![image](https://github.com/robhagemans/pcbasic/assets/157108548/4f57701f-fcdc-41ad-9603-9fd727b4d4dc)

**Reply by BMtech118** [reply]
When i load the file and then list, then edit line 15 to be IF A$<>"N" OR A$<>"n" OR A$<>"J" THEN GOTO 12 which i think should work it saves as Save"xxx.bas but when i then load that file and list the contents again it still has the original line 15 so defo doing it wrong

**Reply by robhagemans** [reply]
Are you able to share the original file at all? It's difficult to find out what's going on from screenshots.

**Reply by BMtech118** [reply]
I can but i can't publicly put it here im afraid.

**Reply by robhagemans** [reply]
When you edited the line, did you press enter? Just changing on the screen won't store the file, you need to press enter while the cursor is on the line you edited.

The file looks regular to me, the erroneous double operators are there in the original file and I don't see anything else that's peculiar. I don't think it would run like this on GW BASIC though I haven't tried yet - there may be some odd special case that makes it run there
